### PR TITLE
Add Jest tests for StaticFilters

### PR DESCRIPTION
**Pull request by nmanu1**
This PR adds Jest tests for the higher-level `StaticFilters` component.

J=SLAP-2111
TEST=auto

Check that the added Jest tests pass.

### CODE REVIEW

File: src/components/Filters/CheckboxOption.tsx [hunk 1]
```diff
@@ -105,7 +105,7 @@ export function CheckboxOption(props: CheckboxOptionProps): JSX.Element | null {
       return false;
     }
 
-    if (!answersUtilities.isCloseMatch(label.toString(), searchValue)) {
```
Comment (oshi97) [inline src/components/Filters/CheckboxOption.tsx:108]: 🔥 🔥 🔥

File: tests/components/StaticFilters.test.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  it('Clicking a filter option executes a search when searchOnChange is true', () => {
+    const actions = spyOnActions();
+    render(<StaticFilters {...staticFiltersProps} />);
+
+    const martyCheckbox: HTMLInputElement = screen.getByLabelText('MARTY!');
+    userEvent.click(martyCheckbox);
+    expect(actions.executeVerticalQuery).toBeCalled();
+  });
+
+  it('Only clicking the apply button executes a search when searchOnChange is false', () => {
```
Comment (oshi97) [inline tests/components/StaticFilters.test.tsx:166]: "only clicking" might be a little unclear, I read it as "I only need to click the apply button to search" as opposed to "the only way I am able to search is by clicking the apply button"

**Review by oshi97 — COMMENTED** [review]
nice tests!

File: tests/components/StaticFilters.test.tsx (hunk 2 above)
Comment (nmanu1) [reply tests/components/StaticFilters.test.tsx]: updated!

**Review by yen-tt — COMMENTED** [review]
Looks like changes from the baked in StaticFilters pr caused some potential errors with AppliedFilters. Specifically, when you applied an option from StaticFilters, that option is displayed twice in AppliedFilters.

**Comment by nmanu1** [comment]
> Looks like changes from the baked in StaticFilters pr caused some potential errors with AppliedFilters. Specifically, when you applied an option from StaticFilters, that option is displayed twice in AppliedFilters.

This is just a bad example of using `StaticFilters` in the test-site. We don't support filtering on the same `fieldId` across multiple components (between multiple `StaticFilters` components or between `Facets` and `StaticFilters`). So the behavior is a little weird when we filter on puppy preference because it's applying the static filter and then also applying the corresponding facet option filter, so 2 identical applied filters are added. I updated the test-site `StaticFilters` to filter on a different `fieldId` to only test behavior we support

**Review by oshi97 — APPROVED** [review]

**Pull request merged**